SimTKcommon/include/SimTKcommon/internal/common.h

```diff
@@ -796,21 +796,86 @@ typedef Is64BitHelper<Is64BitPlatform>::Result Is64BitPlatformType;
 
 /** Attempt to demangle a type name as returned by typeid.name(), with the
 result hopefully suitable for meaningful display to a human. Behavior is 
-compiler-dependent. **/
-SimTK_SimTKCOMMON_EXPORT std::string demangle(const char* name);
-
-/** In case you don't like the name you get from typeid(), you can specialize
-this class to provide a nicer name. This class is typically used for error 
-messages and testing. **/
+compiler-dependent. 
+@relates SimTK::NiceTypeName **/
+SimTK_SimTKCOMMON_EXPORT 
+std::string demangle(const char* name);
+
+/** Given a compiler-dependent demangled type name string as returned by 
+SimTK::demangle(), attempt to form a canonicalized representation that will be
+the same for any compiler. Unnecessary spaces and superfluous keywords like
+"class" and "struct" are removed. The `namestr()` method of NiceTypeName\<T>
+uses this function to produce a human-friendly type name that is the same on any
+platform. The input argument is left empty. 
+@relates SimTK::NiceTypeName **/
+SimTK_SimTKCOMMON_EXPORT 
+std::string canonicalizeTypeName(std::string&& demangledTypeName);
+
+/** Same, but takes an lvalue reference so has to copy the input. 
+@relates SimTK::NiceTypeName **/
+inline std::string canonicalizeTypeName(const std::string& demangledTypeName)
+{   return canonicalizeTypeName(std::string(demangledTypeName)); }
+
+/** Given a canonicalized type name, produce a modified version that is 
+better-suited to use as an XML attribute. This means replacing the angle
+brackets with curly braces to avoid trouble. The input argument is left
+empty. 
+@relates SimTK::NiceTypeName **/
+SimTK_SimTKCOMMON_EXPORT
+std::string encodeTypeNameForXML(std::string&& canonicalizedTypeName);
+
+/** Same, but takes an lvalue reference so has to copy the input. 
+@relates SimTK::NiceTypeName **/
+inline std::string encodeTypeNameForXML(const std::string& niceTypeName)
+{   return encodeTypeNameForXML(std::string(niceTypeName)); }
+
+/** Given a type name that was encoded for XML by SimTK::encodeTypeNameForXML(),
+restore it to its canonicalized form. This means replacing curly braces by
+angle brackets. The input argument is left empty. 
+@relates SimTK::NiceTypeName **/
+SimTK_SimTKCOMMON_EXPORT
+std::string decodeXMLTypeName(std::string&& xmlTypeName);
+
+/** Same, but takes an lvalue reference so has to copy the input. 
+@relates SimTK::NiceTypeName **/
+inline std::string decodeXMLTypeName(const std::string& xmlTypeName)
+{   return decodeXMLTypeName(std::string(xmlTypeName)); }
+
+/** Obtain human-readable and XML-usable names for arbitrarily-complicated
+C++ types. Three methods `name()`, `namestr()`, and `xmlstr()` are provided
+giving respectively the raw, compiler-dependent output from `typeid(T).%name()`, 
```

Comment:
The comma between `raw` and `compiler` is not necessary and makes it seem like there are 4 items in the list.